cloudinit/config/cc_apt_configure.py

```diff
@@ -6,225 +6,364 @@
+          ftp_proxy: 'ftp://[[user][:pass]@]host[:port]/'
+          https_proxy: 'https://[[user][:pass]@]host[:port]/'
+          sources:
+              source1:
+                  keyid: 'keyid'
+                  keyserver: 'keyserverurl'
+                  source: 'deb http://<url>/ xenial main'
+              source2:
+                  source: 'ppa:<ppa-name>'
+              source3:
+                  source: 'deb $MIRROR $RELEASE multiverse'
+                  key: |
+                      ------BEGIN PGP PUBLIC KEY BLOCK-------
+                      <key data>
+                      ------END PGP PUBLIC KEY BLOCK-------""")],
+    'frequency': frequency,
+    'type': 'object',
+    'properties': {
+        'apt': {
+            'type': 'object',
+            'additionalProperties': False,
+            'properties': {
+                'preserve_sources_list': {
+                    'type': 'boolean',
+                    'description': dedent("""\
+                        By default, cloud-init will generate a new sources
+                         list in ``/etc/apt/sources.list.d`` based on any
+                         changes specified in cloud config. To disable this
+                         behavior and preserve the sources list from the
+                         pristine image, set ``preserve_sources_list``
+                         to ``true``.
+
+                        The ``preserve_sources_list`` option overrides
+                         all other config keys that would alter
+                         ``sources.list`` or ``sources.list.d``,
+                         **except** for additional sources to be added
+                         to ``sources.list.d``.""")
+                },
+                'disable_suites': {
+                    'type': 'array',
+                    'items': {
+                        'type': 'string'
+                    },
+                    'uniqueItems': True,
+                    'description': dedent("""\
+                        Entries in the sources list can be disabled using
+                         ``disable_suites``, which takes a list of suites
+                         to be disabled. If the string ``$RELEASE`` is
+                         present in a suite in the ``disable_suites`` list,
+                         it will be replaced with the release name. If a
+                         suite specified in ``disable_suites`` is not
+                         present in ``sources.list`` it will be ignored.
+                         For convenience, several aliases are provided for
+                         ``disable_suites``:
+
+                             ``updates`` => ``$RELEASE-updates``,
+                             ``backports`` => ``$RELEASE-backports``.
+                             ``security`` => ``$RELEASE-security``,
+                             ``proposed`` => ``$RELEASE-proposed``,
+                             ``release`` => ``$RELEASE``.
```

Comment:
Each of these lines in sphinx minimally needs a leading  hyphen '- ' to ensure that they are rendered in readthedocs as bullet items in a list
See https://cloudinit.readthedocs.io/en/latest/topics/modules.html#apt-configure "disable apt sources"
versus your local rendered docs.
To render your docs locally to test and view the differences:
tox -e doc # which generates docs/rtd_html/index.html    
you can browse locally with your browser of choice pointing at the URL like:
file:///home/<your-user>/src/cloud-init/doc/rtd_html/topics/modules.html#apt-configure


Additionally, something in the schema rendering is damaging the multi-line white space that should exist between
```markdown
``disable_suites``:

   - ``updates`` => ``$RELEASE-updates``,
````

So, we'll have to track that down because all this text is being ultimately rendered as a single wrapped line instead of separate bullet points.

Reply:
Yes, I believe the root cause of this issue is caused by this line of [code](https://github.com/canonical/cloud-init/blob/master/cloudinit/config/schema.py#L321). When we do that to the description, we throw away all of the code formatting of lists, notes and multi-lines white spaces. I can address this issue without a problem, but do you think I should work on it on this PR or a separate one ?